### PR TITLE
new handling of the pixel window function, this way planck projected …

### PR DESCRIPTION
**Pull request by thibautlouis**
…to CAR is not assumed to have the CAR window function

### CODE REVIEW

**Comment by thibautlouis** [comment]
This pull request change how the pixel window function is dealt with in the code.  
Again we had to make the dictfile a bit more complicated, we ask the user to define which pixel window function need to be deconvolved for eaach survey. The idea is to have a better handling of case where map have been projected from a pixellisation to an other. For example, when Planck is projected into ACT, you need to deconvolve the Planck pixel window function but not the ACT one.

File: project/data_analysis/python/get_spectra_from_alms.py [hunk 1]
```diff
@@ -89,10 +89,14 @@
     
     xtra_pw1, xtra_pw2, mm_tf1, mm_tf2 = None, None, None, None
     if deconvolve_pixwin:
-        # some xtra pixel window in the case of healpix
-        # or Planck projected into CAR
-        xtra_pw1 = transfer_function.healpix_pixwin(sv1, templates[sv1], binning_file, lmax)
-        xtra_pw2 = transfer_function.healpix_pixwin(sv2, templates[sv2], binning_file, lmax)
+        if d[f"pixwin_{sv1}"]["pix"] == "HEALPIX":
```
Comment (xgarrido) [inline project/data_analysis/python/get_spectra_from_alms.py:92]: what's going on if someone asks for pixwin deconvolution but set `pix` value to CAR ? There are no default value for `xtra_pw{1,2}` parameter (in the `get_alms.py` file, `inv_pixwin_lxly` is `None` by default)

File: project/data_analysis/python/get_alms.py [hunk 2]
```diff
@@ -52,12 +52,12 @@
...
             filter = kspace.get_kspace_filter(win_T, ks_f)
                     
-        if (deconvolve_pixwin == True):
-            # deconvolve the CAR pixel function in fourier space
-            wy, wx = enmap.calc_window(win_T.data.shape, order=d["pixwin_order"])
-            inv_pixwin_lxly = (wy[:,None] * wx[None,:]) ** (-1)
-        else:
-            inv_pixwin_lxly = None
+        inv_pixwin_lxly = None
+        if deconvolve_pixwin == True:
```
Comment (xgarrido) [inline project/data_analysis/python/get_alms.py:56]: you should remove the `== True`

File: project/data_analysis/python/get_spectra_from_alms.py (hunk 1 above)
Comment (thibautlouis) [reply project/data_analysis/python/get_spectra_from_alms.py:92]: you have the line:     xtra_pw1, xtra_pw2, mm_tf1, mm_tf2 = None, None, None, None
then if you pass None to 

 lb, ps = transfer_function.deconvolve_xtra_tf(lb,
                                                                            ps,
                                                                          spectra,
                                                                         xtra_pw1=xtra_pw1,
                                                                        xtra_pw2=xtra_pw2,
                                                                         mm_tf1=mm_tf1,
                                                                   mm_tf2=mm_tf2)

nothing happen. the CAR pixwin deconvolution happen in get_alm, where it's done in map space using a FFT.

**Comment by xgarrido** [comment]
Fix #126

**Pull request merged**